pkg/meta/base.go

```diff
@@ -1561,7 +1562,14 @@ func (m *baseMeta) Close(ctx Context, inode Ino) syscall.Errno {
 	return 0
 }
 
-func (m *baseMeta) Readdir(ctx Context, inode Ino, plus uint8, entries *[]*Entry) syscall.Errno {
+func (m *baseMeta) Readdir(ctx Context, inode Ino, plus uint8, entries *[]*Entry) (rerr syscall.Errno) {
+	defer func() {
+		if rerr == 0 {
+			if err := m.en.touchAtime(ctx, inode); err != 0 {
+				logger.Warnf("readdir %v update atime: %s", inode, err)
+			}
+		}
+	}()
```

Comment:
use the attr got in the later `m.GetAttr`

Reply:
Fixed